### PR TITLE
[#114385219] Upgrade Concourse - bootstrap concourse

### PR DESCRIPTION
**Pull request by timmow**
**What**

A new version of Concourse has been released, we should update to latest version.

**How this PR should be reviewed**

Destroy any existing deployment or use a new environment name

Check this branch out and run the `vagrant/deploy.sh` script, and run the `create-deployer` and `destroy-deployer` pipelines and check all is working.

To make this work, we had to upgrade the kernel on the concourse vm, as the AMI concourse provide uses an older kernel version that doesnt support the new version of the garden linux release included. This means we also need to make the additional disk mount persist across reboots. There are more details on this in the individual commit messages.

Concourse are planning on [upgrading the AMIs](https://www.pivotaltracker.com/n/projects/1059262/stories/114892073) however its not clear when they will be upgraded.

**Who should review this PR**

* Anyone  except me or @mtekel

### CODE REVIEW

File: vagrant/destroy.sh [hunk 1]
```diff
@@ -22,3 +22,4 @@ fi
 
 # 3. Destroy bootstrap concourse
 vagrant destroy -f
+rm .kernel-updated
```
Comment (alext) [inline vagrant/destroy.sh]: This won't cover the case when the machine is deleted by the `self-terminate` pipeline.

File: concourse/pipelines/create-deployer.yml [hunk 2]
```diff
@@ -79,7 +79,8 @@ jobs:
     - get: paas-cf
     - task: try-fetch-bucket-state
       config:
-        image: docker:///governmentpaas/awscli
+        platform: linux
+        image: docker:///ubuntu
```
Comment (dcarley) [inline concourse/pipelines/create-deployer.yml]: There was a comment on an individual commit that hasn't been addressed yet. Making a note here, so we don't forget it: https://github.com/alphagov/paas-cf/commit/c30bb834dd01efeea29b47aa165e57dcac4aa499#diff-5eac5d6fc041d1da10144e10c0fa2ed0L82

File: vagrant/post-deploy.d/20-setup-basic-auth.sh [hunk 3]
```diff
@@ -2,6 +2,7 @@
 set -u
 set -e
 echo "Setting up concourse basic auth as ${CONCOURSE_ATC_USER}: ${CONCOURSE_ATC_PASSWORD}"
+sed "s/--external-url.*/--external-url \'\' \\\/" -i /var/vcap/jobs/atc/bin/atc_ctl # Remove hardcoded auth redirect IP
```
Comment (dcarley) [inline vagrant/post-deploy.d/20-setup-basic-auth.sh]: This sounds like a bug from the description in the commit message. Can you create an issue on the upstream project?

Comment (mtekel) [inline vagrant/post-deploy.d/20-setup-basic-auth.sh]: _what_ upstream project? We only think this AMI is coming from https://github.com/concourse/concourse-lite

Apparently they have some jenkins job that builds the AMI. They have said earlier that AMI is not supported. I am not sure the same issue is present in the lite version - probably not, as this IP could be coming from there.

Comment (mtekel) [inline vagrant/post-deploy.d/20-setup-basic-auth.sh]: But still https://github.com/concourse/concourse-lite/issues/4

Comment (dcarley) [inline vagrant/post-deploy.d/20-setup-basic-auth.sh]: Thanks.

For the benefit of anyone else following along: they have explicitly stated in that issue that they don't intend to build the AMI any longer.

**Comment by dcarley** [comment]
We decided not to proceed with this because the workarounds required outweighed the benefit of upgrading the Bootstrap Concourse, where we aren't using as many features and don't use as frequently. It has been superseded by [#115322155](https://www.pivotaltracker.com/story/show/115322155) to make sure that we use the correct version of `fly` against each Concourse. We'll need another story, in the near future, to look at alternatives to using the upstream AMI which is no longer supported.

**Pull request closed without merging**